// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kuscia.proto.api.v1alpha1;

import "google/protobuf/any.proto";

option go_package = "github.com/secretflow/kuscia/proto/api/v1alpha1";
option java_package = "org.secretflow.v1alpha1.common";

// RequestHeader carries the user custom headers.
message RequestHeader {
  // Custom headers used to record custom information.
  map<string, string> custom_headers = 1;
}

// Status carries the response status information.
// Reference: https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto
message Status {
  // The status code, which should be one of google rpc code or custom code.
  int32 code = 1;
  // Message for recording the error information.
  string message = 2;
  // A list of messages that carry the additional supplementary error details.
  repeated google.protobuf.Any details = 3;
}

message Partition {
    // enum path, odps
	string   type = 1;
	repeated DataColumn fields = 2;
}

// DataColumn defines the column of data.
message DataColumn {
	string name = 1;
	// enum: int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64, date32, date64, bool, string,
	// binary
	string type = 2;
	// The description of column
	string comment = 3;
	// can the column could be nullable, default is nullable
	bool not_nullable = 4;
}

enum FileFormat {
	UNKNOWN = 0;
	CSV     = 1;
	BINARY  = 2;
}
